### PR TITLE
Add computed padding for handling clipped large Net worth amounts

### PR DESCRIPTION
**Pull request by sreetamdas**
In the Net Worth report, amounts over ~6 digits get clipped because of [zero margin added](https://github.com/actualbudget/actual/blob/08aff05a06f4bca4344f939324080e1fa6511b62/packages/desktop-client/src/components/reports/graphs/NetWorthGraph.tsx#L124).

```jsx
<AreaChart
  width={width}
  height={height}
  data={graphData.data}
  margin={{ top: 0, right: 0, left: 0, bottom: 0 }}
>
```

<details><summary>Before</summary>
<p>

image

</p>
</details> 

---

Fix by adding a left padding that takes into account the length of the net worth amount (so it can handle even bigger amounts, which is common in currencies like INR, VND, etc.)

```jsx
/**
 * Add left padding for Y-axis for when large amounts get clipped
 */
function computePadding(netWorthData: Array<{ y: number }>) {
  /**
   * Convert to string notation, get longest string length
   */
  const maxLength = Math.max(
    ...netWorthData.map(({ y }) => {
      return Math.round(y).toLocaleString().length;
    }),
  );

  if (maxLength <= 5) {
    // No additional left padding is required for upto 5 characters
    return 0;
  }

  return (maxLength - 5) * 5;
}
```

<details><summary>After</summary>
<p>

image

</p>
</details>

### CODE REVIEW

File: packages/desktop-client/src/components/reports/graphs/NetWorthGraph.tsx [hunk 1]
```diff
@@ -180,3 +185,26 @@ export function NetWorthGraph({
...
+      return Math.round(y).toLocaleString().length;
+    }),
+  );
+
+  if (maxLength <= 5) {
+    // No additional left padding is required for upto 5 characters
+    return 0;
+  }
+
+  return (maxLength - 5) * 5;
```
Comment (psybers) [inline packages/desktop-client/src/components/reports/graphs/NetWorthGraph.tsx:209]:
```suggestion
  // No additional left padding is required for upto 5 characters
  return Math.max(0, (maxLength - 5) * 5);
```

**Review by psybers — CHANGES_REQUESTED** [review]

File: packages/desktop-client/src/components/reports/graphs/NetWorthGraph.tsx (hunk 1 above)
Comment (sreetamdas) [reply packages/desktop-client/src/components/reports/graphs/NetWorthGraph.tsx]: This is better indeed, thanks!

**Review by psybers — APPROVED** [review]

**Comment by psybers** [comment]
You will need to update the VRT tests.

File: packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-net-worth-and-cash-flow-reports-1-chromium-linux.png
Comment (psybers) [inline packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-net-worth-and-cash-flow-reports-1-chromium-linux.png:1]: It looks like there might be a bit of a problem.  The net worth graph is now slightly cut off on the left side.

**Review by psybers — CHANGES_REQUESTED** [review]

File: packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-net-worth-and-cash-flow-reports-1-chromium-linux.png
Comment (sreetamdas) [reply packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-net-worth-and-cash-flow-reports-1-chromium-linux.png:1]: Sorry about that @psybers, I've fixed this and run VRT again and updated the snapshots. Thanks!

**Review by psybers — CHANGES_REQUESTED** [review]

File: packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-net-worth-and-cash-flow-reports-1-chromium-linux.png
[the remark this answers is not in the document]
Comment (sreetamdas) [reply packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-net-worth-and-cash-flow-reports-1-chromium-linux.png:1]: Sorry, should be fixed now!

**Review by psybers — APPROVED** [review]

**Comment by youngcw** [comment]
Im guessing that merging master will fix these conflicts.  Can you look into that?

**Review by youngcw — APPROVED** [review]
Thanks!

**Pull request merged**